rotkehlchen/chain/optimism/node_inquirer.py

```diff
@@ -50,32 +50,37 @@ def __init__(
             msg_aggregator=database.msg_aggregator,
         )
         contracts = EvmContracts[Literal[ChainID.OPTIMISM]](chain_id=ChainID.OPTIMISM)
-        super().__init__(
+        DSProxyInquirerWithCacheData.__init__(
+            self,
             greenlet_manager=greenlet_manager,
             database=database,
             etherscan=etherscan,
             blockchain=SupportedBlockchain.OPTIMISM,
             etherscan_node=OPTIMISM_ETHERSCAN_NODE,
             etherscan_node_name=OPTIMISM_ETHERSCAN_NODE_NAME,
             contracts=contracts,
-            rpc_timeout=rpc_timeout,
             contract_multicall=contracts.contract(string_to_evm_address('0x2DC0E2aa608532Da689e89e237dF582B783E552C')),  # noqa: E501
             contract_scan=contracts.contract(string_to_evm_address('0x1e21bc42FaF802A0F115dC998e2F0d522aDb1F68')),  # noqa: E501
             dsproxy_registry=contracts.contract(string_to_evm_address('0x283Cc5C26e53D66ed2Ea252D986F094B37E6e895')),  # noqa: E501
             native_token=A_ETH.resolve_to_crypto_asset(),
+            rpc_timeout=rpc_timeout,
+        )
+        OptimismSuperchainInquirer.__init__(
```

Comment:
Hm I see what you meant by your earier question in discord. It's suboptimal the way it is now. As you are inheriting like this:

```
                EvmNodeInquirer
                       |
      +----------------+------------------+
      |                                    |
EvmNodeInquirerWithDSProxy      OptimismSuperchainInquirer
      |                                    |
      |                                    |
DSProxyInquirerWithCacheData               |
      |                                    |
      |                                    |
      |                                    |
      |---------------------------- OptimismInquirer
      |                                    
      |
 EthereumInquirer
```

And as such the OptimismInquirer gets to re-initialize all fields two times and have duplication.

The only solution I can think of now, which is cleaner, but still not super idea is to:

- Add another abstract class `DSProxyOptimismSuperchainInquirerWithCacheData`. Damn that's a loooong name :/
- Have it inherit from SuperChainNodeInquirer, add the DSProxy and Cache stuff and make it abstract.
- Then inherit from that here, so that no duplication happens.